PROTOCOL.md

```diff
@@ -141,8 +141,10 @@ Field    | Type | Description
 
 ## Control Layer
 
-All messages of the Control Layer are transmitted as JSON arrays with the following fields: `[version, type, ...typeSpecificFields]`. Each message of the Control Layer is a payload of the [WrapperMessage](#wrappermessage) defined in the Network Layer.
-`version` describes the version of the Control Layer. `type` is an integer to identify the message type according to the following table: 
+This document describes version `2` of the Control Layer protocol.
+
+All messages of the Control Layer are transmitted as JSON arrays with the following fields: `[version, type, requestId, ...typeSpecificFields]`. Each message of the Control Layer is a payload of the [WrapperMessage](#wrappermessage) defined in the Network Layer.
+`version` describes the version of the Control Layer. `type` is an integer to identify the message type according to the below table. `requestId` is a string that gets echoed back in the response(s). It should be unique within the same connection. 
```

Comment:
👍

Reply:
Maybe could provide some guidance as to how to generate these ids and what kind of behaviour should/could be expected if it encounters reused ids?

Reply:
If I understand correctly, `requestId` should be set to be a random string when sending messages that fall under the section "Requests sent".  And then messages under the section "Responses sent" just copy the value of `requestId` from the corresponding request. This is sort of implied here but could be explained in more detail to make sure it is clear.

Reply:
Clarified it more. Just to set it straight among us:
- It doesn't need to be random, just unique within the connection
- Yes responses echo back the `requestId` value of the request they're responding to

> what kind of behaviour should/could be expected if it encounters reused ids

- Reusing `requestId`s within the connection is a violation of the protocol. Protocol specs need to specify the protocol, but usually don't describe what happens if someone doesn't implement the protocol correctly.